### PR TITLE
Changing json library

### PR DESCRIPTION
**Pull request by koufopoulosf**
Some performance improvement with orjson library.

### CODE REVIEW

File: snscrape/modules/twitter.py [hunk 1]
```diff
@@ -32,7 +32,8 @@
 import typing
 import urllib.parse
 import warnings
-
+import orjson
+json = orjson
```
Comment (TheTechRobo) [inline snscrape/modules/twitter.py:36]: Better to use `import orjson as json`, or better yet, replace all references to `json` with `orjson`.

**Review by TheTechRobo — APPROVED** [review]
`orjson` needs to be installed beforehand, it's not preinstalled. Please add it to the setup.py as a dependency.

Also, [there are more modules than just the twitter scraper that use JSON](https://github.com/JustAnotherArchivist/snscrape/search?q=import+json). Do you want to update all of them?

**Comment by TheTechRobo** [comment]
Do you have any benchmarks?

**Comment by koufopoulosf** [comment]
@TheTechRobo Thanks for your answer.

> Do you have any benchmarks?

From my experience, json serialization is always a major bottleneck and thus a faster json library is always a must.

Watching the code more carefully, there are indeed a lot more code junks that need to be changed or removed completely.
For example you can see here:
https://github.com/JustAnotherArchivist/snscrape/blob/0d361685ff77425aaf7cc6c6580237d914041a55/snscrape/base.py#L29

While https://github.com/ijl/orjson has already solved most of these things, the whole snscrape needs to be synced with orjson library for better understanding the performance gains that can potentially have. For 1000 tweets I get 1 second improvement, (by using this wrong json = orjson line), however json library seems wired to the code and it requires further look into this.

**Comment by JustAnotherArchivist** [comment]
I'm aware of other JSON packages and their performance benefits. (I'm actually the co-maintainer of one.) I'm absolutely open to adding support for using them in snscrape. This is not how it should be integrated though. Requirements I can think of right now:

* No hard dependency on any of them. Fall back to stdlib `json` if they are not installed.
* Used throughout snscrape for parsing and encoding, probably through a helper module that handles the importing etc.
* Identical JSONL output regardless of the encoder used. For example, some encoders have different default values for `separators`. This should be handled in the helper module, not by passing `separators` in every call. The rest of the code should use it as if stdlib `json` were in use.

I never benchmarked it in snscrape. I know that the HTML parsing used to be by far the slowest part (apart from the network fetches at times, but that's irrelevant), but at least the Twitter module basically no longer parses HTML anyway.

**Comment by koufopoulosf** [comment]
@JustAnotherArchivist A very fast html parser I've worked before is this one:

https://github.com/rushter/selectolax

I hope it helps.

**Comment by JustAnotherArchivist** [comment]
I've been thinking about this, and the best approach would probably be a separate package for that JSON indirection so that it's reusable outside of snscrape's narrow context. There are a couple already (anyjson and omnijson), but they haven't been maintained in a decade, and they don't handle those encoder differences I mentioned. So a new package is probably in order. Maybe some day... Until then, the standard library will have to do.

**Pull request closed without merging**